### DISCUSSION TITLE
C0 data

### DISCUSSION THREAD

**Discussion by minmfang** [hse-project/hse#36] [Q&A] [answered]
If there is a problem with the power supply, will the data in C0 be lost ？ （have not been flushed to Cn)

**Comment by smoyerx** [comment, marked as answer]
Yes, data in c0 will be lost.  The KVDB parameter dur_intvl_ms controls the max time data is cached before automatically flushing to media.  And there are API calls for forcing cached data to media.

**Comment by alkkkj** [comment]
Is it still true in HSE 2.0? I noticed some WAL code in the source tree. Not sure whether used to keep data if power off.

**Comment by smoyerx** [comment]
Yes, HSE 2.0 still caches data in c0.  When that data is flushed to stable storage (now a WAL) is determined by the [durability controls](https://hse-project.github.io/2.x/dev/concepts/#durability-controls) as documented.
